pkg/apis/mongodb/v1/mongodb_types.go

```diff
@@ -118,6 +118,9 @@ type MongoDBUser struct {
 
 	// Roles is an array of roles assigned to this user
 	Roles []Role `json:"roles"`
+
+	// ScramCredentialsSecretName appended by string "scram-credentials" is the name of the secret object created by the mongoDB operator for storing SCRAM credentials
+	ScramCredentialsSecretName string `json:"scramCredentialsSecretName"`
```

Comment:
we should ensure that for auto generation of the CRD we include the pattern here

```golang
	// ScramCredentialsSecretName appended by string "scram-credentials" is the name of the secret object created by the mongoDB operator for storing SCRAM credentials
	// +kubebuilder:validation:Pattern="^[a-z0-9]([-a-z0-9]*[a-z0-9])?(\.[a-z0-9]([-a-z0-9]*[a-z0-9])?)*$"
	ScramCredentialsSecretName string `json:"scramCredentialsSecretName"`
```
(we're currently only not doing this due to the arbitrary StatefulSet generation, you can test this yourself with the command `operator-sdk generate crds --crd-version v1beta1)

Reply:
done